### PR TITLE
Events Calendar + pull events from Google Calendar

### PR DESCRIPTION
**Pull request by RobbieZhuang**
Events page and home page events are linked to the [SE Soc calendar](https://calendar.google.com/calendar?cid=bGI2dHRiMWtzMzdxbDVyOGlzdWFzZ2NkbzhAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ). **Currently, there are no upcoming events so the events list will be empty.**

To load calendar events from a google calendar
1. create a `.env` file in the root dir `public-website`
2. Get **Calendar ID** from google calendar and set `REACT_APP_GOOGLE_CALENDAR_ID` to it
3. Get **[Google API Key](https://developers.google.com/calendar/quickstart/js)** from google calendar and set `REACT_APP_GOOGLE_CALENDAR_API_KEY` to it

**Desktop Calendar View**
Screen Shot 2020-04-28 at 2 51 57 PM
**Desktop Day View**
Screen Shot 2020-04-28 at 3 32 48 PM

**Mobile Calendar View**
Screen Shot 2020-04-28 at 3 34 03 PM

TODO:
- ~~Filter events list to only display upcoming on events page list & home page list~~
- ~~Fix CSS on calendar component~~
- ~~Create custom toolbar~~
- ~~Clean up module/gcal and streamline the two separate calls to `getEvents()` from Home and Events pages.~~
   - Keep the two separate calls to `getEvents()`, maybe consider Redux in the future.
- ~~Add official SE Soc calendar in~~

### CODE REVIEW

File: src/containers/Events.jsx [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
+          </Col>
+          <Col sm={7}>
+            <div style={{height: "600px"}}>
+              <EventCalendar events={this.state.events} />
+            </div>
+            <div className="add-cal">
+              <Button
+                variant="secondary"
+                className="add-cal-btn"
+                href="https://calendar.google.com/calendar/ical/9lt5o1quln7c4u5d4mhm223pa4%40group.calendar.google.com/public/basic.ics"
```
Comment (RobbieZhuang) [inline src/containers/Events.jsx:39]: This link will download the calendar .ics (idk what type of access level it is)

Comment (jenniferwills) [reply src/containers/Events.jsx]: unfortunately I don't think .ics files update with new events, they just contain info from the current events on the calendar. Can you use ```https://calendar.google.com/calendar?cid=bGI2dHRiMWtzMzdxbDVyOGlzdWFzZ2NkbzhAZ3JvdXAuY2FsZW5kYXIuZ29vZ2xlLmNvbQ``` instead? This should open google calendar and give the option to add the calendar to your google calendars (read only access).

Comment (RobbieZhuang) [reply src/containers/Events.jsx]: I like your method better :) & changed it to the link you provided.

fyi, from my testing, it seems like the ics does update with new events.

Comment (jenniferwills) [reply src/containers/Events.jsx]: the ICS does update with the current events when you download a new one, but if you have it downloaded and then I add a new event, you need to re-download the ICS to get the new events. It is a one time pull, not a subscription. It just contains a copy of all existing events from the calendar, and they are yours to edit/delete etc but changes won't be reflected to anyone else's versions, and you won't automatically get any changes made to the original events

Comment (RobbieZhuang) [reply src/containers/Events.jsx]: Ah you're right - I was _already subscribed_ to the calendar I was editing. I thought I was editing the ics one when in reality it wasn't, my bad 😅

File: src/components/styles/ReactBigCalendar.scss [hunk 2]
```diff
@@ -0,0 +1,741 @@
+@charset "UTF-8";
```
Comment (RobbieZhuang) [inline src/components/styles/ReactBigCalendar.scss:1]: This is a copy of the react-big-calendar styles with customized styling.

File: src/containers/styles/Events.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
+$color-eng: #692D90;
+$color-eng-light: #b496c7;
+$color-eng-lightest: #e1d5e8;
+.container {
```
Comment (tawnyzhao) [inline src/containers/styles/Events.scss:4]: Setting `.container` here will affect all containers on the site, instead set the class or id of the container in `event.jsx` or use bootstrap's spacing classes: https://getbootstrap.com/docs/4.0/utilities/spacing/

File: src/modules/gcal.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import Axios from "axios";
+import Moment from "moment";
+
+const CALENDAR_ID = process.env.REACT_APP_GOOGLE_CALENDAR_ID;
+const API_KEY = process.env.REACT_APP_GOOGLE_CALENDAR_API_KEY;
```
Comment (jenniferwills) [inline src/modules/gcal.js:5]: These will need to be populated on the deployment machine, correct? I'll make a note on that issue #31 to make sure to add them

File: src/containers/styles/Events.scss (hunk 3 above)
Comment (RobbieZhuang) [reply src/containers/styles/Events.scss]: Good call, thanks.

**Comment by RobbieZhuang** [comment]
> We should consider pulling all of the colors out into a `colors.scss` and importing.

Moved colours to colors.scss & imported where used.

File: src/modules/gcal.js (hunk 4 above)
Comment (RobbieZhuang) [reply src/modules/gcal.js:5]: Yes! Thanks for doing so :).

**Pull request merged**